opentreemap/treemap/js/src/mapPage/locationSearchUI.js

```diff
@@ -60,12 +78,46 @@ function setPolygon(newPolygon) {
     polygon = newPolygon.addTo(map);
 }
 
-function clearCustomArea() {
-    if (polygon) {
-        map.removeLayer(polygon);
-        polygon = null;
-    }
+function completePolygon(newPolygon) {
+    var lngLats = _.map(newPolygon.getLatLngs()[0], function (p) {
+            return [p.lng, p.lat];
+        }),
+        ring = lngLats.concat([_.take(lngLats)]),
+        data = {polygon: ring},
+        anonymousBoundaryStream = createAnonymousBoundary(data)
+            .map(function (result) {
+                return result.id;
+            });
+
+    setPolygon(newPolygon);
+    anonymousBoundaryStream.onValue(function (boundaryID) {
+        $(dom.locationValue).val(boundaryID);
+        boundarySelect.shouldZoomOnLayerChange(false);
+    });
+
+    // Map id from anonymousBoundaryStream into the data structure
+    // that locationChangeBus expects.
+    var builtSearchStream = Bacon.combineTemplate({
+        display: null,
+        filter: {
+            'mapFeature.geom': {IN_BOUNDARY: anonymousBoundaryStream}
+        }
+    });
```

Comment:
This suggestion results in the issue we saw before where `editAreaMode` shows the edit tooltip but not the markers. I don't know why yet. Investigating.

There's another bug here. Without your suggested changes, when editing a polygon in `editDrawMode`, if you click "Search" instead of "Done", although the location search box returns to the "_CUSTOM AREA_" display, clicking the pencil icon no longer gets into edit area mode at all.

Investigating that, too.